src/ngraph/pass/reshape_elimination.cpp

```diff
@@ -192,3 +191,127 @@ void pass::ReshapeElimination::construct_dot_transpose_pattern()
+    auto rehape_label = make_shared<pattern::op::Label>(reshape, no_fan_out, NodeVector{reshape});
+
+    auto callback = [op, rehape_label](pattern::RecurrentMatcher& m) {
+        NGRAPH_DEBUG << "In callback for construct_recurrent_reshape against node = "
+                     << rehape_label->get_argument(0)->get_name();
+        auto reshape_node_vector = m.get_bound_nodes_for_pattern(rehape_label);
+
+        // The bounded node vector is in reverse order. It is convinient to have the
+        // bounded node vector in the correct order
+        std::reverse(std::begin(reshape_node_vector), std::end(reshape_node_vector));
+
+        auto first_bounded_reshape_op = reshape_node_vector.front();
+        auto driver_op = first_bounded_reshape_op->get_argument(0);
+        auto last_bounded_reshape_op = reshape_node_vector.back();
+
+        // Need to check if the next to last bounded op is a reshape since the last reshape is allowed
+        // to have fan-out but the matcher will discard any reshape if it has fan-out
+        auto user_of_last_bounded_reshape_op = last_bounded_reshape_op->get_users(true)[0];
+        if (std::dynamic_pointer_cast<op::Reshape>(user_of_last_bounded_reshape_op))
+        {
+            reshape_node_vector.push_back(user_of_last_bounded_reshape_op);
+            last_bounded_reshape_op = reshape_node_vector.back();
+        }
+
+        // Return if the recurrent matcher matches only one reshape
+        if (reshape_node_vector.size() == 1)
+        {
+            return false;
+        }
+
+        // The complete reshape node vector may not contain contigous reshapes that can be
+        // fused. Only the subset of reshapes with a reshape(any axis order) followed by reshapes
+        // with default axis order can be fused. Creating such subpatterns here:
+        std::vector<NodeVector> sub_patterns{NodeVector{first_bounded_reshape_op}};
+        for (auto it = std::next(reshape_node_vector.begin()); it != reshape_node_vector.end();
+             it++)
+        {
+            auto r = std::dynamic_pointer_cast<op::Reshape>(*it);
+
+            // Check that the input to r is the last reshape stored in the
+            // subpattern vector
+            if (!r)
+            {
+                NGRAPH_DEBUG
+                    << "Incorrect match. Something went wrong. Non-reshape op has been matched";
+                return false;
+            }
+
+            auto default_order_r = get_default_order(r->get_input_shape(0));
+            if (r->get_input_order() == default_order_r)
+            {
+                sub_patterns.back().push_back(r);
+            }
+            else
+            {
+                NGRAPH_DEBUG << r->get_name() << "does not have default axis order. "
+                             << "It might be part of a different subpattern";
+                sub_patterns.push_back(NodeVector{r});
+            }
+        }
+
+        // Remove the subpatterns with just one reshape in them
```

Comment:
Can we just use `continue` in the loop that follows, rather than modify the vector here?

Reply:
True..I just overcomplicated this . :\